cmd/scan/scan.go

```diff
@@ -0,0 +1,220 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	_ "log"
+	"os"
+	"path/filepath"
+
+	"strconv"
+
+	"github.com/btcsuite/btcd/rpcclient"
+	"github.com/btcsuite/btcutil"
+)
+
+type Address struct {
+	Addr         string `json:"address"`
+	MinScanBlock int64  `json:"min_scan_block"`
+	MidScanBlock int64  `json:"mid_scan_block"`
+	MaxScanBlock int64  `json:"max_scan_block"`
+	Txs          []Tx   `json:"txs"`
+}
+
+type Tx struct {
+	TxHash      string `json:"tx_hash"`
+	BlockHash   string `json:"block_hash"`
+	ParentHash  string `json:"parent_hash"`
+	BlockHeight int64  `json:"block_height"`
+}
+
+type Deposit struct {
+	Addr string
+	Tx   Tx
+}
+
+func ScanBlock(client *rpcclient.Client, blockID int64) ([]Deposit, error) {
+	blockHash, err := client.GetBlockHash(blockID)
+	if err != nil {
+		return nil, err
+	}
+	block, err := client.GetBlockVerboseTx(blockHash)
+	if err != nil {
+		return nil, err
+	}
+	parentHash := block.PreviousHash
+	var deposits []Deposit
+	var depTx Tx
+	depTx.BlockHeight = blockID
+	depTx.BlockHash = blockHash.String()
+	depTx.ParentHash = parentHash
+	for _, tx := range block.RawTx {
+		for i, addr := range tx.Vout {
+			depTx.TxHash = tx.Txid
+			depTx.TxHash = depTx.TxHash + ":" + strconv.Itoa(i)
+			if len(addr.ScriptPubKey.Addresses) > 0 {
+				deposits = append(deposits, Deposit{Addr: addr.ScriptPubKey.Addresses[0], Tx: depTx})
+			}
+		}
+
+	}
+	return deposits, nil
+}
+
+func CompareAddress(addr Address, deps []Deposit) Address {
+	for _, dep := range deps {
+		if addr.Addr == dep.Addr {
+			if addr.Txs[0].BlockHeight == -1 {
+				addr.Txs[0] = dep.Tx
+			} else {
+				if !ExistTx(addr, dep.Tx) {
+					addr.Txs = append(addr.Txs, dep.Tx)
+				}
+			}
+		}
+	}
+
+	return addr
+}
+
+func ExistTx(addr Address, tx Tx) bool {
+	for _, t := range addr.Txs {
+		if t == tx {
+			return true
+		}
+	}
+	return false
+}
+
+func UpdateAddressInfo(addrs []Address, deps []Deposit, blockID int64) []Address {
+	branch := 0
+	for i, addr := range addrs {
+		branch = 0
+
+		if addr.MaxScanBlock == 0 && branch == 0 && blockID > 1 {
```

Comment:
you can use `if else` or a `switch` instead of `branch` flag

Reply:
done